src/libs/elektra/opmphm.c

```diff
@@ -8,8 +8,90 @@
 
 
 #include <kdbassert.h>
+#include <kdblogger.h>
 #include <kdbopmphm.h>
 
+#include <stdio.h>  //debug
+#include <string.h> //strlen
+#include <time.h>   //time
+
+/* Prints the bipartite graph to a dot file, to draw it, for debug purpose. */
+void opmphmPrintGraph (Edge * edges, void ** data, opmphmGetString fpOpmhpmGetString, size_t n)
+{
+	FILE * f = fopen ("opmphmGraph.dot", "wr");
+	fprintf (f, "graph graphname {\n");
+	for (size_t i = 0; i < n; ++i)
+	{
+		uint32_t h1 = edges[i].h[1];
+		uint32_t h2 = edges[i].h[2];
+		fprintf (f, "%u -- %u [label=\"%s\"]\n", h1, h2, fpOpmhpmGetString (data[i]));
+	}
+	fprintf (f, "}\n");
+	fclose (f);
+}
+
+// The ratio defines the number of vertices on one side of the biparithegraph, denoted as r.
+// Double this value to get to the Fox et al. ratio.
+const double ratio = 0.75;
+
+size_t opmphmR (size_t n)
+{
+	return n * ratio;
+}
+
+/* The first phase of the order preserving minimal perfect hash map build. */
+int opmphmMapping (OPMPHM * opmphm, Vertex * vertices, Edge * edges, void ** data, opmphmGetString fpOpmhpmGetString, size_t n)
+{
+	unsigned int seed = time (NULL);
+	// set the seeds, for the hash function
+	for (int i = 0; i < 3; ++i)
+	{
+		opmphm->opmphmHashFunctionSeeds[i] = opmphmRandom (&seed);
+	}
+	size_t r = opmphmR (n);
+	ELEKTRA_LOG_DEBUG ("OPMPHM: Mapping: r=%lu seed[0]=%u seed[1]=%u seed[2]=%u\n", r, opmphm->opmphmHashFunctionSeeds[0],
```

Comment:
Isn't this more important then the logging of the individual keys? (Maybe simply use `ELEKTRA_LOG`)?